cg/store/filters/status_illumina_metrics_filters.py

```diff
@@ -0,0 +1,46 @@
+"""Filters for the Illumina Sample Sequencing Metrics objects."""
+
+from enum import Enum
+from typing import Callable
+
+from sqlalchemy.orm import Query
+
+from cg.store.models import (
+    IlluminaFlowCell,
+    IlluminaSampleSequencingMetrics,
+    IlluminaSequencingRun,
+    Sample,
+)
+
+
+def filter_by_run_id_sample_internal_id_and_lane(
+    metrics: Query, run_id: str, sample_internal_id: str, lane: int, **kwargs
+) -> Query:
+    """Filter metrics by run id, sample internal id and lane."""
+    joined_query: Query = metrics.join(Sample).join(IlluminaSequencingRun).join(IlluminaFlowCell)
+    return joined_query.filter(
+        Sample.internal_id == sample_internal_id,
```

Comment:
Im considering turning this into 3 different filter functions and concatenated them. I did this like this originally to mimic the old metrics filtering function but maybe splitting it is better?

Reply:
This will be a really expensive query I think. it joins all three FULL tables. Perhaps first querying for the right samples then joining the reduced tables later might be a better strategy?

So basically the read.py function (i.e. the get) will first reduce the tables then do the join on the reduced tables. 

In the order of:
1. get the Illumina flow cell filtered query
2. Get the sequencing run filtered query
3. get the samples for the samples table filtered query
4. Join all three reduced ones